tests/suites/test_suite_gcm.aes128_de.data

```diff
@@ -670,6 +670,58 @@ AES-GCM NIST Validation (AES-128,128,1024,1024,32) #2 [#2]
 depends_on:MBEDTLS_AES_C
 gcm_decrypt_and_verify:MBEDTLS_CIPHER_ID_AES:"659b9e729d12f68b73fdc2f7260ab114":"fd0732a38224c3f16f58de3a7f333da2ecdb6eec92b469544a891966dd4f8fb64a711a793f1ef6a90e49765eacaccdd8cc438c2b57c51902d27a82ee4f24925a864a9513a74e734ddbf77204a99a3c0060fcfbaccae48fe509bc95c3d6e1b1592889c489801265715e6e4355a45357ce467c1caa2f1c3071bd3a9168a7d223e3":"459df18e2dfbd66d6ad04978432a6d97":"ee0b0b52a729c45b899cc924f46eb1908e55aaaeeaa0c4cdaacf57948a7993a6debd7b6cd7aa426dc3b3b6f56522ba3d5700a820b1697b8170bad9ca7caf1050f13d54fb1ddeb111086cb650e1c5f4a14b6a927205a83bf49f357576fd0f884a83b068154352076a6e36a5369436d2c8351f3e6bfec65b4816e3eb3f144ed7f9":32:"8e5a6a79":"FAIL":"":0
 
+AES-GCM NIST - empty ciphertext, AD length: 128 bytes, ciphertext updates: 0
+depends_on:MBEDTLS_AES_C
+gcm_decrypt_and_verify_empty_cipher:MBEDTLS_CIPHER_ID_AES:"63c3f81500746eaf383fe3975d84f849":"0799d4152fd73c1604b4610cf7171fe1":"cb8248e5f904cc9ccccf6f273fe621eee1b4d7ed98480f9e806a48b84e2d6a733772ecf8fb7fe91805715cddab2b462b89f6e6c7cf873f65031f13c357d5f57b00b7c391c39e78ad1ed94be236ca0ae316bce11bc33c5d701fdfc58abbe918b9c42f7b3d6e89d46f9784b388a6e6daf47730b9fa665d755a17e89932fa669c44":"c53d01e53ee4a6ea106ea4a66538265e":0
+
+AES-GCM NIST - empty ciphertext, AD length: 128 bytes, ciphertext updates: 1
+depends_on:MBEDTLS_AES_C
+gcm_decrypt_and_verify_empty_cipher:MBEDTLS_CIPHER_ID_AES:"63c3f81500746eaf383fe3975d84f849":"0799d4152fd73c1604b4610cf7171fe1":"cb8248e5f904cc9ccccf6f273fe621eee1b4d7ed98480f9e806a48b84e2d6a733772ecf8fb7fe91805715cddab2b462b89f6e6c7cf873f65031f13c357d5f57b00b7c391c39e78ad1ed94be236ca0ae316bce11bc33c5d701fdfc58abbe918b9c42f7b3d6e89d46f9784b388a6e6daf47730b9fa665d755a17e89932fa669c44":"c53d01e53ee4a6ea106ea4a66538265e":1
+
+AES-GCM NIST - empty ciphertext, AD length: 128 bytes, ciphertext updates: 2
+depends_on:MBEDTLS_AES_C
+gcm_decrypt_and_verify_empty_cipher:MBEDTLS_CIPHER_ID_AES:"63c3f81500746eaf383fe3975d84f849":"0799d4152fd73c1604b4610cf7171fe1":"cb8248e5f904cc9ccccf6f273fe621eee1b4d7ed98480f9e806a48b84e2d6a733772ecf8fb7fe91805715cddab2b462b89f6e6c7cf873f65031f13c357d5f57b00b7c391c39e78ad1ed94be236ca0ae316bce11bc33c5d701fdfc58abbe918b9c42f7b3d6e89d46f9784b388a6e6daf47730b9fa665d755a17e89932fa669c44":"c53d01e53ee4a6ea106ea4a66538265e":2
+
+AES-GCM NIST - empty ciphertext, AD length: 90 bytes, ciphertext updates: 0
```

Comment:
Is the reference to NIST correct here ? For the three tests before I can see that there is a corresponding NIST test case before by not for this one. Related question, where does this test data come from? Ditto for the "en" data.

Reply:
Test data comes from http://csrc.nist.gov/groups/STM/cavp/documents/mac/gcmtestvectors.zip

Reply:
Thanks for the reference. Wouldn't it better to say "NIST CAVS 14.0" to identify the source ?

Reply:
It would be nice to identify the source precisely, with the URL, in the commit message. Completely precise identification is a bit too much information in the commit description, which has to fit in 66 characters to be displayed fully.

Reply:
Just asking as I don't really know how it works with NIST and its test vectors. Isn't the case that the URL used here could in the future point to a new version of the test vectors which does not include the test vectors added here anymore?

Reply:
Yes. For an unambiguous reference, you need both the URL and some version information. Some indication of the download date, such as an implicit date of “a few days before the commit”, is good enough. A version number is definitely better when there is one.

Reply:
Should I edit the commit messages? (rebase + force push required)

Reply:
Editing the commit message is not an issue from our reviewer perspective as we can check that the diff of the force-push is empty. Thus please improve the commit message. Otherwise the NIST files you pointed out start with the header:

 CAVS 14.0
# GCM Decrypt with keysize 128 test information
# Generated on Fri Aug 31 11:28:04 2012

CAVS 14.0 seems to be the version and I think it would be a good idea to mention it in the test description, thus:
`AES-GCM NIST CAVS 14.0 - ... `